src/AceEditor/madie-ace-editor.tsx

```diff
@@ -1,19 +1,37 @@
-import React from "react";
+import React, { useEffect, useRef } from "react";
 import AceEditor from "react-ace";
 
-import "ace-builds/src-noconflict/mode-java";
+import "ace-builds/src-noconflict/mode-sql";
 import "ace-builds/src-noconflict/theme-monokai";
+import "ace-builds/src-noconflict/ext-language_tools";
+import CqlMode from "./cql-mode";
 
 export default function MadieAceEditor({ props }) {
+  const aceRef = useRef<AceEditor>(null);
+  const localStorageKey = props.localStorageKey ?? "cql";
+  const [cql, setCql] = React.useState(localStorage.getItem(localStorageKey));
+
+  useEffect(() => {
+    const cqlMode = new CqlMode();
+    // @ts-ignore
+    aceRef?.current?.editor?.getSession()?.setMode(cqlMode);
+    localStorage.setItem(localStorageKey, cql);
+  }, [cql, localStorageKey]);
+
   return (
     <AceEditor
-      mode="java"
+      mode="sql" // Temporary value of mode to prevent a dynamic search request.
```

Comment:
Could you elaborate on the "dynamic search request?"

Reply:
@AndrewBird81  My understanding is that Ace doesn't work well with webpack resources out of the box. It will try to load a js file directly from the server based on the file name and location of other modes. That would show an ERROR in the console. As a workaround, we can provide an existing known mode and then override it with a custom one.